src/endsWith.ts

```diff
@@ -0,0 +1,26 @@
+import { getEq } from 'fp-ts/lib/Array';
+import { Eq } from 'fp-ts/lib/Eq';
+import { takeLast } from './takeLast';
+
+function _endsWith<A>(E: Eq<A>): (suffix: Array<A>, as: Array<A>) => boolean {
+  const eqArray = getEq(E);
+  return (suffix, as) => eqArray.equals(suffix, takeLast(suffix.length, as));
+}
+
+export function endsWith<A>(
+  E: Eq<A>
+): {
+  (suffix: Array<A>): (as: Array<A>) => boolean;
+  (suffix: Array<A>, as: Array<A>): boolean;
+};
+
+export function endsWith<A>(E: Eq<A>): (suffix: Array<A>, oas?: Array<A>) => any {
```

Comment:
minor: `oas` as a name was temporarily introduced in previous PRs to fight against the `no-shadowed-variable` TSLint rule. I think that sticking to `as` here is clearer.

Reply:
You're right. I was going for `as` and `bs` then switched to `suffix` and `list` (used in ramda code). Then @gcanti took it over.